conformance/05_referrers.go

```diff
@@ -0,0 +1,276 @@
+			g.Specify("GET request to nonexistent blob should result in empty 200 response", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(dummyDigest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(BeZero())
+			})
+
+			g.Specify("GET request to existing blob should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).ToNot(BeZero())
+			})
+
+			g.Specify("GET request to existing blob with filter should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest)).
+					SetQueryParam("artifactType", "application/vnd.oci.conformance")
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).ToNot(BeZero())
+
+				// also check resp header "OCI-Filters-Applied: artifactType" denoting that an artifactType filter was applied
+				Expect(resp.Header().Get("OCI-Filters-Applied")).ToNot(BeEmpty())
+				Expect(resp.Header().Get("OCI-Filters-Applied")).To(Equal("application/vnd.oci.conformance"))
+			})
+		})
+
+		g.Context("Error codes", func() {
+			g.Specify("400 response body should contain OCI-conforming JSON message", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference("sha256:totallywrong")).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(invalidManifestContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAny(
+					Equal(http.StatusBadRequest),
+					Equal(http.StatusNotFound)))
+				if resp.StatusCode() == http.StatusBadRequest {
+					errorResponses, err := resp.Errors()
+					Expect(err).To(BeNil())
+
+					Expect(errorResponses).ToNot(BeEmpty())
+					Expect(errorCodes).To(ContainElement(errorResponses[0].Code))
+				}
+			})
```

Comment:
Did we specify this anywhere in the spec? It feels like undefined behavior and I'd be okay with registries either returning a 400 or accepting the manifest without returning a `OCI-Subject` header.